docs/NOW-spec.md

```diff
@@ -396,50 +123,33 @@ The NOW_STRING16 structure is used to represent variable-length strings of up to
     </thead>
     <tbody>
         <tr>
-            <td colspan="8">len</td>
-            <td colspan="24">str (variable)</td>
+            <td colspan="32">msgSize</td>
         </tr>
         <tr>
-            <td colspan="32">...</td>
+            <td colspan="8">msgClass</td>
+            <td colspan="8">msgType</td>
+            <td colspan="16">msgFlags</td>
         </tr>
     </tbody>
 </table>
 
-**len (1 byte)**: An unsigned 8-bit number containing the string length, excluding the null terminator. The maximum value is 15.
-
-**str (variable)**: The UTF-8 encoded string excluding the null terminator.
-
-##### NOW_STRING32
-
-The NOW_STRING32 structure is used to represent variable-length strings of up to 15 characters that can easily fit within a fixed-size buffer of 32 bytes.
+**msgSize (4 bytes)**: The message size, excluding the header size (8 bytes).
 
-<table class="byte-layout">
-    <thead>
-        <tr>
-            <th>0</th><th>1</th><th>2</th><th>3</th><th>4</th><th>5</th><th>6</th><th>7</th>
-            <th>8</th><th>9</th><th>10</th><th>1</th><th>2</th><th>3</th><th>4</th><th>5</th>
-            <th>6</th><th>7</th><th>8</th><th>9</th><th>20</th><th>1</th><th>2</th><th>3</th>
-            <th>4</th><th>5</th><th>6</th><th>7</th><th>8</th><th>9</th><th>30</th><th>1</th>
-        </tr>
-    </thead>
-    <tbody>
-        <tr>
-            <td colspan="8">len</td>
-            <td colspan="24">str (variable)</td>
-        </tr>
-        <tr>
-            <td colspan="32">...</td>
-        </tr>
-    </tbody>
-</table>
+**msgClass (1 byte)**: The message class.
 
-**len (1 byte)**: An unsigned 8-bit number containing the string length, excluding the null terminator. The maximum value is 31.
+| Flag                            | Meaning              |
+|---------------------------------|----------------------|
+| NOW_NEGOTIATION_MSG_CLASS_ID<br>0x10 | Protocol negotiation class |
+| NOW_SYSTEM_MSG_CLASS_ID<br>0x11 | System message class |
+| NOW_SESSION_MSG_CLASS_ID<br>0x12 | Session message class |
+| NOW_EXEC_MSG_CLASS_ID<br>0x13 | Exec message class |
 
-**str (variable)**: The UTF-8 encoded string excluding the null terminator.
+**msgType (1 byte)**: The message type, specific to the message class.
 
-##### NOW_STRING64
+**msgFlags (2 bytes)**: The message flags, specific to the message type and class.
 
-The NOW_STRING64 structure is used to represent variable-length strings of up to 63 characters that can easily fit within a fixed-size buffer of 64 bytes.
+#### NOW_STATUS
```

Comment:
Diff looks messy here, it is better to view `NOW_STATUS` changes in raw file/rendered PDF, however, here is the quick list of changes:
- Simplified NOW_STATUS overall structure, its not based on HRESULT now. 
- It contains a 32-bit code field instead of a 16-bit one; now it could fit the process exit code from any OS.
- Defines common error kinds - now-proto-specific errors, win32 errors, unix errors. Error kinds could be extended.
- Provides optional string message field for errors
- Defines more now-proto error kinds based on DVC server implementation experience
- Removed severity. Not its either success or error status.